src/auxinfo/participant.rs

```diff
@@ -420,8 +485,9 @@ impl AuxInfoParticipant {
         let proof = AuxInfoProof::prove(
             rng,
             self.retrieve_context(),
-            message.id(),
+            sid,
             global_rid,
+            // XXX BUG! this should be the _other party_'s setup parameters!
             my_public.params(),
```

Comment:
I'm very confused why this passes verification: we pass our own setup parameters here, but as far as I can tell we use the other participant's setup parameters when verifying the proof below?

Reply:
This is really hard to follow, I agree. Maybe @tyurek is more familiar with what's going on here?

Reply:
Oh, this does make sense. The sender creates `Pi_fac` using only their own AuxInfo parameters. The verifier than says "OK this proof should verify with the sender's AuxInfo parameters." It's wrong, but it's consistent.

Reply:
Ah yes! That does make sense.

Reply:
I agree with swan. The self/other switch makes sense between prove and verify because different parties run those protocols, but there is still a bug because I think the prover should use the verifier's commitment parameters to create the proof.